packages/twenty-front/src/modules/billing/hooks/useBillingPlan.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  type SubscriptionInterval,
+  useListPlansQuery,
+} from '~/generated-metadata/graphql';
+import { currentWorkspaceState } from '@/auth/states/currentWorkspaceState';
+import { findOrThrow } from '~/utils/array/findOrThrow';
+import {
+  type BillingPlanKey,
+  type BillingPriceLicensedDto,
+  type BillingPriceMeteredDto,
+  BillingProductKey,
+  BillingUsageType,
+} from '~/generated/graphql';
+import { type MeteredBillingPrice } from '@/billing/types/billing-price-tiers.type';
+import { isDefined } from 'twenty-shared/utils';
+import { useRecoilValue } from 'recoil';
+
+export const useBillingPlan = () => {
```

Comment:
this hook is problematic to me:
- it's an umbrella hook doing many things, this means that using it will trigger a re-render nightmare!
- all these getters are not idiomatic to me: the issue is that these are functions and that we use the function return in react which is new for react engine at each re-renders and therefore can trigger a re-render itself

I would recommend creating hooks (maybe not one for everything but more and to return the things directly
For instance:
usePlans()
useProducts()

useProduct(planKey)
etc...